qml/datahandler.cpp

```diff
@@ -19,35 +19,28 @@
 
 #include "datahandler.h"
 
+#include <QtCore/QMap>
 #include <QtCore/QDebug>
 #include <QtDBus/QDBusReply>
 #include <QtDBus/QDBusInterface>
 #include <QtDBus/QDBusConnection>
 
-DataHandler::DataHandler(QObject* parent): QObject(parent)
-{
-    QDBusConnection dbus = QDBusConnection::sessionBus();
-    QDBusInterface* interface = new QDBusInterface("org.kde.kdenow", "/KDENow");
+DataHandler::DataHandler(QObject* parent): QObject(parent) {
 
-    //Call a method, to start the kdenowd daemon if it hasn't yet started
-    QDBusReply<QString> reply = interface->call("startDaemon");
-    if (reply.isValid()) {
-        qDebug() << "Valid Reply received from org.kde.kdenow /KDENow";
-        qDebug() << reply.value();
+    connect(this, &DataHandler::addedToWallet, this, &DataHandler::setDBusConnections);
+    connect(this, &DataHandler::showUserCredentialsPage, this,
+            &DataHandler::onShowUserCredentialsPage);
+    m_wallet = KWallet::Wallet::openWallet("KDENowWallet",
+                                            0,
+                                            KWallet::Wallet::Synchronous
+                                          );
+    if (!m_wallet->hasFolder("KDENow")) {
+        emit showUserCredentialsPage();
     }
     else {
-        qDebug() << "Did not receive a valid reply from org.kde.kdenow /KDENow";
-        return;
+        emit addedToWallet();
```

Comment:
Although logically code is correct but emit signal name doesn't comply with it (Nothing has been added to wallet).